### ISSUE TITLE
log in not responding on 2.0.5-RC2

### ISSUE THREAD

**Issue by childers** [GMOD/Apollo#1371]
Testing 2.0.5-RC2 in my BLAST -> JBrowse -> Apollo workflow, and everything looks ok until I try logging in.  It hangs until it throws a 502 timeout error.  

After logging in separately and watching the network logger in the browser, it shows one request (the initial GET request).

Doing the same request while not logged in loads the browser in JBrowse mode fine. Logging in at this point I see this in the tomcat logs:
```shell
0:0:0:0:0:0:0:1 - - [06/Dec/2016:11:41:46 -0500] "GET /apollo-2.0.5-SNAPSHOT/assets/application-74401703485edfb4d46c11fb4531561b.css HTTP/1.0" 200 25685
127.0.0.1 - - [06/Dec/2016:11:41:56 -0500] "POST /apollo-2.0.5-SNAPSHOT/auth/signIn HTTP/1.0" 302 -
```
And it just hangs until it throws a timeout error.

**Comment by nathandunn** [comment]
@childers   Make sure that you have cleared the cache.   I'm going to double-check that its not a routing error.  

From a public jbrowse:

<img width="534" alt="screen shot 2016-12-06 at 8 52 04 am" src="https://cloud.githubusercontent.com/assets/751274/20934899/a72e923a-bb91-11e6-928f-db93096ed4cc.png">


Hit login button on the far right:

<img width="1433" alt="screen shot 2016-12-06 at 8 55 21 am" src="https://cloud.githubusercontent.com/assets/751274/20934956/d0e875fa-bb91-11e6-86ad-11da02f349fc.png">

Enter credentials and hit login:

<img width="824" alt="screen shot 2016-12-06 at 8 55 30 am" src="https://cloud.githubusercontent.com/assets/751274/20934955/d0e37550-bb91-11e6-93e6-d6b906c732bd.png">

Logged in:

<img width="1054" alt="screen shot 2016-12-06 at 8 55 49 am" src="https://cloud.githubusercontent.com/assets/751274/20934954/d0e01dba-bb91-11e6-9605-b911c07cb1a1.png">

**Comment by nathandunn** [comment]
Adding the additional URL's, I do get a weird error, though:

    http://icebox.lbl.gov/Apollo-staging/Honeybee/jbrowse/index.html?addTracks=[{%22label%22:%22BLAST%20%20Results%22,%22category%22:%220.%20Reference%20Assembly%22,%22type%22:%22WebApollo/View/Track/DraggableBLASTFeatures%22,%22store%22:%22url%22,%22style%22:{%22renderClassName%22:%22gray-center-10pct%22,%22subfeatureClasses%22:{%22match_part%22:%22blast-match_part%22}}}]&addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3D&addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev%3D%5Bnal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C+nal%3A%5Busda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C+usda%3A%5Bgov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C+gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul%3A%5Bscaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C+scaffolds%3A%5B50_new_ids.fa.gff%22%7D%7D%3A%2C+50_new_ids%3A%5Bfa.gff%22%7D%7D%3A%2C+fa%3A%5Bgff%22%7D%7D%3A%5D%5D%5D%5D%5D%5D%5D&loc=Scaffold1186:145061..145979=&loc=Scaffold1186:145061=[.145979:,%20:[145979:]]&tracks=BLAST++Results=&loc=Group1.25:39977..399010&tracks=DNA,Annotations,BLAST%20%20Results&highlight=&

redirects to here adding a second Apollo-staging (grr):

  http://icebox.lbl.gov/Apollo-staging/Apollo-staging/annotator/index?clientToken=280441480471316317392&addTracks=[{%22label%22:%22BLAST%20%20Results%22,%22category%22:%220.%20Reference%20Assembly%22,%22type%22:%22WebApollo/View/Track/DraggableBLASTFeatures%22,%22store%22:%22url%22,%22style%22:{%22renderClassName%22:%22gray-center-10pct%22,%22subfeatureClasses%22:{%22match_part%22:%22blast-match_part%22}}}]&addStores=[Ljava.lang.String;@1326d4fe&tracks=[Ljava.lang.String;@5427b6ab&highlight=

**Comment by nathandunn** [comment]
Related to #1214.    I am seeing the "blank screen" with the 500 error locally as well when it goes to "Load Link".  

from here it is fine:

   http://localhost:8080/apollo/Honeybee/jbrowse/index.html?addTracks=%5B%7B%22label%22%3A%22BLAST%20%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3D=&addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev%3D%5Bnal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bnal%3A%5Busda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Busda%3A%5Bgov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bgov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul%3A%5Bscaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bscaffolds%3A%5B50_new_ids.fa.gff%22%7D%7D%3A%2C%2B50_new_ids%3A%5Bfa.gff%22%7D%7D%3A%2C%2Bfa%3A%5Bgff%22%7D%7D%3A%5D%5D%5D%5D%5D%5D%5D=&loc=Group1.1%3A1..214200&tracks=DNA%2CAnnotations%2CBLAST%20%20Results&highlight=

clicking on login goes:

    http://localhost:8080/apollo/annotator/loadLink?addTracks=%5B%7B%22label%22%3A%22BLAST%20%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3D=&addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev%3D%5Bnal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bnal%3A%5Busda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Busda%3A%5Bgov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bgov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul%3A%5Bscaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bscaffolds%3A%5B50_new_ids.fa.gff%22%7D%7D%3A%2C%2B50_new_ids%3A%5Bfa.gff%22%7D%7D%3A%2C%2Bfa%3A%5Bgff%22%7D%7D%3A%5D%5D%5D%5D%5D%5D%5D=&loc=Group1.1%3A1..214200&tracks=DNA%2CAnnotations%2CBLAST%20%20Results&highlight=&organism=Honeybee&clientToken=18711201645223582151849581675

which is a 500 error.   Note that in the above "working" version addStores and &tracks are String objects, which might be why it was working, though it is still a different type of error.

**Comment by childers** [comment]
Oh loadLink is the gift that keeps on giving!

**Comment by nathandunn** [comment]
Modified as addStores was added twice:

   http://localhost:8080/apollo/Honeybee/jbrowse/index.html?addTracks=%5B%7B%22label%22%3A%22BLAST%20%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev%3D%5Bnal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bnal%3A%5Busda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Busda%3A%5Bgov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bgov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul%3A%5Bscaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bscaffolds%3A%5B50_new_ids.fa.gff%22%7D%7D%3A%2C%2B50_new_ids%3A%5Bfa.gff%22%7D%7D%3A%2C%2Bfa%3A%5Bgff%22%7D%7D%3A%5D%5D%5D%5D%5D%5D%5D=&loc=Group1.1%3A1..214200&tracks=DNA%2CAnnotations%2CBLAST%20%20Results&highlight=

**Comment by nathandunn** [comment]
test string:

   http://localhost:8080/apollo/Honeybee/jbrowse/?loc=Scaffold59:274204..275731&addStores={%22url%22:{%22type%22:%22JBrowse/Store/SeqFeature/GFF3%22,%22urlTemplate%22:%22https://some-server/webapp/media/blast/task/46d547d45f8042b3910edf5b9d08665d/Aros01112013-genome_new_ids.fa.gff%22}}&addTracks=[{%22label%22:%22BLAST+%20Results%22,%22category%22:%220.%20Reference%20Assembly%22,%22type%22:%22WebApollo/View/Track/DraggableBLASTFeatures%22,%22store%22:%22url%22,%22style%22:{%22renderClassName%22:%22gray-center-10pct%22,%22subfeatureClasses%22:{%22match_part%22:%22blast-match_part%22}}}]&tracks=BLAST+%20Results

**Comment by childers** [comment]
This is not fixed.  The request URL is getting mangled, I think it is not recognizing the FQDN, as we have subdomains for the different services (in production we have i5k.nal.usda.gov passing dataStores to the Apollo app on apollo.nal.usda.gov, and analogous systems for testing new releases).  the Request URL looks to only have the main domain, pointing the request to a different machine that doesn't have Apollo. I can get redirected to Apollo when not logged in without issue and view in non-curation mode, but once I click login it seems to get lost and hang.

See the image below for an example.  


<img width="1427" alt="2 0 5-snapshot-error" src="https://cloud.githubusercontent.com/assets/421248/20941453/5ccb365c-bbc4-11e6-99de-2d30f35aba5f.png">

**Comment by monicacecilia** [comment]
@childers FYI merging the PR closes the issue automatically. Further testing still needed. Thanks keeping a close eye.

**Comment by nathandunn** [comment]
verified it is a proxy issue:

```
http://icebox.lbl.gov/Apollo-staging/auth/login?targetUri=/annotator/loadLink%3Fov%2FApollo-staging%2FHoneybee%2Fjbrowse%2F%3Floc%3DGroup1.1%3A1..214200%26addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F46d547d45f8042b3910edf5b9d08665d%2FAros01112013-genome_new_ids.fa.gff%22%7D%7D%26addTracks%3D%5B%7B%22label%22%3A%22BLAST%2B+Results%22%2C%22category%22%3A%220.+Reference+Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D%26tracks%3DDNA%2CAnnotations%26highlight%3D%26organism%3DHoneybee%26clientToken%3D1316033811627232621701398713%26controller%3Dannotator%26format%3Dnull%26action%3DloadLink
```

**Comment by nathandunn** [comment]
Going through local tomcat is not a problem, but if I use a reverse proxy it runs into issues . . . fix forthcoming

**Comment by nathandunn** [comment]
problem found with nginx . . . having issues getting the proxy to work locally though (keeps killing the server, which is bizarre)

**Comment by nathandunn** [comment]
Going from : 

   http://icebox.lbl.gov/Apollo-staging/Honeybee/jbrowse/?loc=Group1.1%3A1..214200&addStores=%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F46d547d45f8042b3910edf5b9d08665d%2FAros01112013-genome_new_ids.fa.gff%22%7D%7D&addTracks=%5B%7B%22label%22%3A%22BLAST%2B%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&tracks=DNA%2CAnnotations&highlight=

to:

   http://icebox.lbl.gov/Apollo-staging/auth/login?targetUri=/annotator/loadLink%3Fov%2FApollo-staging%2FHoneybee%2Fjbrowse%2F%3Floc%3DGroup1.1%3A1..214200%26addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F46d547d45f8042b3910edf5b9d08665d%2FAros01112013-genome_new_ids.fa.gff%22%7D%7D%26addTracks%3D%5B%7B%22label%22%3A%22BLAST%2B+Results%22%2C%22category%22%3A%220.+Reference+Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D%26tracks%3DDNA%2CAnnotations%26highlight%3D%26organism%3DHoneybee%26clientToken%3D1316033811627232621701398713%26controller%3Dannotator%26format%3Dnull%26action%3DloadLink

so expected: 

    targetUri=/annotator/loadLink?loc=Group1.1:1..214200 . . . 

but got:

    targetUri=/annotator/loadLink%3Fov%2FApollo-staging%2FHoneybee%2Fjbrowse%2F%3Floc%3DGroup1.1.214200 . . . 

so: "/" is being resolved relatively.

**Comment by nathandunn** [comment]
weird .. now I'm getting this string:

   ?highlight=&addTracks=[{"label":"BLAST+ Results","category":"0. Reference Assembly","type":"WebApollo/View/Track/DraggableBLASTFeatures","store":"url","style":{"renderClassName":"gray-center-10pct","subfeatureClasses":{"match_part":"blast-match_part"}}}]&organism=cow&clientToken=17961775261757518441&addStores=%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F46d547d45f8042b3910edf5b9d08665d%2FAros01112013-genome_new_ids.fa.gff%22%7D%7D&ov/Apollo-staging/cow/jbrowse/?loc=AY526085.1:1..16338&tracks=DNA,Annotations,BLAST+ Results&format=null 

which is formatted backwards from what I would expect and would fail.

**Comment by nathandunn** [comment]
Had issues with this with nginx . . . somehow the assets got confused:

```
LoadModule proxy_module libexec/apache2/mod_proxy.so 
LoadModule proxy_html_module libexec/apache2/mod_proxy_html.so 

LoadModule proxy_http_module libexec/apache2/mod_proxy_http.so 
LoadModule proxy_wstunnel_module libexec/apache2/mod_proxy_wstunnel.so 
...
 ProxyRequests Off  
 ProxyPass /apollo http://localhost:8080/apollo
 ProxyPassReverse /apollo http://localhost:8080/apollo
 ProxyPass /apollo/stomp  ws://localhost:8080/apollo/stomp
 ProxyPassReverse /apollo/stomp ws://localhost:8080/apollo/stomp
```

**Comment by nathandunn** [comment]
Not a problem with Apache proxy config (though this stomp config is not exactly right above):

<img width="1368" alt="screen shot 2016-12-07 at 11 14 58 am" src="https://cloud.githubusercontent.com/assets/751274/20982694/7586dae8-bc6e-11e6-8b3a-63f90bd3dba2.png">

hit "login" yields:

   http://localhost/apollo/auth/login?targetUri=/annotator/loadLink?highlight=&addTracks=[{%22label%22:%22BLAST+%20Results%22,%22category%22:%220.%20Reference%20Assembly%22,%22type%22:%22WebApollo/View/Track/DraggableBLASTFeatures%22,%22store%22:%22url%22,%22style%22:{%22renderClassName%22:%22gray-center-10pct%22,%22subfeatureClasses%22:{%22match_part%22:%22blast-match_part%22}}}]&organism=Honeybee&clientToken=774268534791328348824068847&addStores=%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F46d547d45f8042b3910edf5b9d08665d%2FAros01112013-genome_new_ids.fa.gff%22%7D%7D&tracks=DNA,Annotations,BLAST+%20Results&ollo/Honeybee/jbrowse/?loc=Group1.1:1..214200&format=null

<img width="1436" alt="screen shot 2016-12-07 at 11 16 18 am" src="https://cloud.githubusercontent.com/assets/751274/20982745/a3ba5fb6-bc6e-11e6-8d39-e91f53b8eb6f.png">

**Comment by nathandunn** [comment]
going to retry with nginx proxy on AWS.

**Comment by nathandunn** [comment]
this is a build on master BTW.

**Comment by nathandunn** [comment]
Working with nginx using.  Note: if you install with nginx with homebrew, use sudo to run it (**not** sudo nginx services start).

```
http {
# . . . 
  #gzip  on;
    map $http_upgrade $connection_upgrade {
        default upgrade;
        ''      close;
    }


   server {
        listen       80;
        server_name  localhost;

# . . . 
 location /apollo {
                               #include         /etc/nginx/proxy.conf;
               proxy_redirect          off;

               proxy_set_header        Host            $host;
               proxy_set_header        X-Real-IP       $remote_addr;
               proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;

               #               proxy_set_header        Host            $host;
               #               proxy_set_header        X-Real-IP       $remote_addr;
               #               proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
               client_max_body_size    10m;
               client_body_buffer_size 128k;
               proxy_connect_timeout   90;
               proxy_send_timeout      360;
               proxy_read_timeout      360;
               proxy_buffers           32 4k;

               proxy_http_version 1.1;
               proxy_set_header Upgrade $http_upgrade;
               proxy_set_header Connection $connection_upgrade;
               proxy_pass      http://127.0.0.1:8080/apollo;
#             proxy_pass      http://localhost:8080;
           }
   }
}
```

**Comment by nathandunn** [comment]
Using grails run-app . . was not able to reproduce.   Now, I'm going to try using run-war and then tomcat8 with an installed war.

**Comment by nathandunn** [comment]
Using nginx 1.10 on icebox

**Comment by nathandunn** [comment]
As a note, WRT to [this configuration](https://github.com/GMOD/Apollo/issues/1371#issuecomment-265555230), I tried both version of the proxy-pass (with and without the trailing apollo flag) and they both work.

**Comment by nathandunn** [comment]
test URL (logged out):

    http://localhost/apollo/Honeybee/jbrowse/?loc=Group1.1%3A1..214200&addStores=%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F46d547d45f8042b3910edf5b9d08665d%2FAros01112013-genome_new_ids.fa.gff%22%7D%7D&addTracks=%5B%7B%22label%22%3A%22BLAST%2B%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&tracks=DNA%2CAnnotations&highlight=

**Comment by nathandunn** [comment]
Got it to reproduce the error using nginx and tomcat 8 (going back to reproduce again).   The the problem is that when it goes into the load link, instead of ```?loc=XXXX``` it does -> ```&80/apollo/Honeybee/jbrowse/?loc=Group1.1:1..214200&```

    http://localhost/apollo/auth/login?targetUri=/annotator/loadLink?highlight=&addTracks=[{%22label%22:%22BLAST+%20Results%22,%22category%22:%220.%20Reference%20Assembly%22,%22type%22:%22WebApollo/View/Track/DraggableBLASTFeatures%22,%22store%22:%22url%22,%22style%22:{%22renderClassName%22:%22gray-center-10pct%22,%22subfeatureClasses%22:{%22match_part%22:%22blast-match_part%22}}}]&organism=Honeybee&clientToken=6124472522553347306669&addStores=%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F46d547d45f8042b3910edf5b9d08665d%2FAros01112013-genome_new_ids.fa.gff%22%7D%7D&tracks=DNA,Annotations&ollo/Honeybee/jbrowse/?loc=Group1.1:707352..742203&format=null

**Comment by nathandunn** [comment]
On further review, am able to get this in dev-mode (yeah!) as well, hopefully a solution soon.

**Comment by nathandunn** [comment]
In SecurityFilters, one of the param keyvalues coming into the security filter is:

    "ollo/Honeybee/jbrowse/?loc" -> "Group1.1:707352..742203"

Basically, ?loc should never show up as a key . . . fixed that and it should be working.

**Comment by nathandunn** [comment]
Got this error: 

    {"error":"No signature of method: static java.net.URLEncoder.encode() is applicable for argument types: (org.codehaus.groovy.grails.web.servlet.mvc.GrailsParameterMap, java.lang.String) values: [[nal.usda.gov/webapp/media/blast/task/8a3be3bab6fa45eb8b7a2e2b298daf41/Cscul.scaffolds.50_new_ids.fa.gff\"}}=:, ...], ...]\nPossible solutions: encode(java.lang.String), encode(java.lang.String, java.lang.String), encodeAsJs(), decodeJs(), hashCode()"}

did not get locally, will fix and recompile (not sure why it would break).

**Comment by nathandunn** [comment]
from: 

    http://icebox.lbl.gov/Apollo-staging/Honeybee/jbrowse/index.html?addTracks=%5B%7B%22label%22%3A%22BLAST%20%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3D=&addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev%3D%5Bnal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bnal%3A%5Busda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Busda%3A%5Bgov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bgov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul%3A%5Bscaffolds.50_new_ids.fa.gff%22%7D%7D%3A%2C%2Bscaffolds%3A%5B50_new_ids.fa.gff%22%7D%7D%3A%2C%2B50_new_ids%3A%5Bfa.gff%22%7D%7D%3A%2C%2Bfa%3A%5Bgff%22%7D%7D%3A%5D%5D%5D%5D%5D%5D%5D=&loc=Group1.15%3A593578..613746&tracks=DNA%2CAnnotations%2CBLAST%20%20Results%2CGeneID&highlight=

**Comment by nathandunn** [comment]
Give me a minute . . here @monicacecilia

**Comment by nathandunn** [comment]
yeah, now I'm getting a 500 error, but no error report  . . @#%#@

**Comment by nathandunn** [comment]
when securityfilter goes to                                 

    redirect(uri: "/auth/login?targetUri=${targetUri}")

somehow it returns a 500, but with no response . . . probably an error on the proxy side then.

**Comment by nathandunn** [comment]
Is ready and fixed by #1374
